sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/Filter.java

```diff
@@ -84,95 +87,128 @@
 
   /** Implementation of the filter. */
   public static class Inner<T> extends PTransform<PCollection<T>, PCollection<T>> {
-    private final Map<String, SerializableFunction<?, Boolean>> fieldNameFilters =
-        Maps.newHashMap();
-    private final Map<Integer, SerializableFunction<?, Boolean>> fieldIdFilters = Maps.newHashMap();
-    private final Map<List<String>, SerializableFunction<Row, Boolean>> fieldNamesFilters =
-        Maps.newHashMap();
-    private final Map<List<Integer>, SerializableFunction<Row, Boolean>> fieldIdsFilters =
-        Maps.newHashMap();
+    @AutoValue
+    abstract static class FilterDescription<FieldT> implements Serializable {
+      abstract FieldAccessDescriptor getFieldAccessDescriptor();
+
+      abstract SerializableFunction<FieldT, Boolean> getPredicate();
+
+      @Nullable
+      abstract Schema getOutputSchema();
+
+      abstract boolean getUnbox();
```

Comment:
Nit: same as above, maybe `selectedSingleField` or `accessedSingleField`?

Reply:
Done

Reply:
Thank you for the changes! LGTM.